thirdparty/CMakeLists.txt

```diff
@@ -134,106 +134,54 @@ if(ENABLE_ONEDNN_FOR_GPU)
-                BUILD_BYPRODUCTS ${ONEDNN_OUTPUT_LIBRARY}
-            )
-            
-        endif()
-
-        ExternalProject_Get_property(onednn_gpu_build SOURCE_DIR)
-        ExternalProject_Get_property(onednn_gpu_build BINARY_DIR)
-
-        # WA: create temp 'include' directory
-        file(MAKE_DIRECTORY "${BINARY_DIR}/include")
-
-        if (NOT isMultiConfig)
-            add_library(onednn_gpu STATIC IMPORTED GLOBAL)
-            set_target_properties(onednn_gpu PROPERTIES
-                IMPORTED_LOCATION ${ONEDNN_OUTPUT_LIBRARY}
-                INTERFACE_INCLUDE_DIRECTORIES "$<BUILD_INTERFACE:${BINARY_DIR}/include>;$<BUILD_INTERFACE:${SOURCE_DIR}/include>"
-                INTERFACE_COMPILE_DEFINITIONS ENABLE_ONEDNN_FOR_GPU)
-            add_dependencies(onednn_gpu onednn_gpu_build)
-        endif()
-
-        # TODO: does not work for imported targets; need to import this target in
-        # OpenVINOConfig.cmake explicitly
-        # ov_install_static_lib(onednn_gpu gpu)
+        ExternalProject_Add(onednn_gpu_build
+            SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/onednn_gpu"
+            BINARY_DIR "${ONEDNN_BUILD_DIR}"
+            INSTALL_DIR "${ONEDNN_INSTALL_DIR}"
+            PREFIX "${ONEDNN_PREFIX_DIR}"
+            EXCLUDE_FROM_ALL ON
+            CMAKE_ARGS
+                "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
+                "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+                "-DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER}"
+                "-DCMAKE_C_COMPILER_LAUNCHER=${CMAKE_C_COMPILER_LAUNCHER}"
+                "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}"
+                "-DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}"
+                "-DCMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE=${ENABLE_LTO}"
+                "-DCMAKE_POLICY_DEFAULT_CMP0069=NEW"
+                "-DDNNL_CPU_RUNTIME=NONE"
+                "-DDNNL_GPU_RUNTIME=OCL"
+                "-DDNNL_LIBRARY_NAME=onednn_gpu"
+                "-DCMAKE_INSTALL_PREFIX=${ONEDNN_INSTALL_DIR}"
+                "-DCMAKE_INSTALL_LIBDIR=lib/$<CONFIG>"
+                "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+                "-DDNNL_ENABLE_CONCURRENT_EXEC=ON"
+                "-DDNNL_ENABLE_PRIMITIVE_CACHE=ON"
+                "-DDNNL_ENABLE_JIT_PROFILING=${BUILD_SHARED_LIBS}"
+                "-DDNNL_ENABLE_ITT_TASKS=${BUILD_SHARED_LIBS}"
+                "-DDNNL_BUILD_TESTS=OFF"
+                "-DDNNL_BUILD_EXAMPLES=OFF"
+                "-DDNNL_BLAS_VENDOR=NONE"
+                "-DDNNL_LIBRARY_TYPE=STATIC"
+                "-DOpenCL_LIBRARY=${OpenCL_LIBRARY}"
+                "-DOpenCL_INCLUDE_DIR=${OpenCL_INCLUDE_DIR}"
+                "-DOPENCL_VERSION_2_2=${OPENCL_VERSION_2_2}"
+        )
+        add_library(onednn_gpu_tgt INTERFACE)
+        set_target_properties(onednn_gpu_tgt PROPERTIES
+            INTERFACE_LINK_DIRECTORIES "${ONEDNN_INSTALL_DIR}/lib/$<CONFIG>"
```

Comment:
does it require cmake 3.21?

Reply:
I tested with 3.16 and 3.18. Documentation says that this propery is available since 3.13

Reply:
I mean usage of generator expression in this property